src/org/jitsi/impl/neomedia/rtp/translator/SimulcastController.java

```diff
@@ -0,0 +1,533 @@
+            {
+                if (!RTPPacketPredicate.INSTANCE.test(pkts[i]))
+                {
+                    continue;
+                }
+
+                long srcSSRC = pkts[i].getSSRCAsLong();
+                if (srcSSRC != state.currentSSRC)
+                {
+                    // We do _not_ forward packets from SSRCs other than the
+                    // current SSRC.
+                    pkts[i] = null;
+                    continue;
+                }
+
+                int srcSeqNum = pkts[i].getSequenceNumber();
+                int dstSeqNum = state.rewriteSeqNum(srcSeqNum);
+
+                long srcTs = pkts[i].getTimestamp();
+                long dstTs = state.rewriteTimestamp(srcTs);
+
+                if (logger.isDebugEnabled())
+                {
+                    logger.debug("src_ssrc=" + pkts[i].getSSRCAsLong()
+                        + ",src_seq=" + srcSeqNum
+                        + ",src_ts=" + srcTs
+                        + ",dst_ssrc=" + targetSSRC
+                        + ",dst_seq=" + dstSeqNum
+                        + ",dst_ts=" + dstTs);
+                }
+
+                if (srcSeqNum != dstSeqNum)
+                {
+                    pkts[i].setSequenceNumber(dstSeqNum);
+                }
+
+                if (dstTs != srcTs)
+                {
+                    pkts[i].setTimestamp(dstTs);
+                }
+
+                if (srcSSRC != targetSSRC)
+                {
+                    pkts[i].setSSRC((int) targetSSRC);
+                }
+            }
+
+            return pkts;
+        }
+    }
+
+    /**
+     * Transforms outgoing RTCP packets.
+     */
+    private class RTCPTransformer implements PacketTransformer
+    {
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void close()
+        {
+            // TODO update the receive counters
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] reverseTransform(RawPacket[] pkts)
+        {
+            //  No need for that (yet?).
+            return pkts;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public RawPacket[] transform(RawPacket[] pkts)
+        {
+            if (ArrayUtils.isNullOrEmpty(pkts))
+            {
+                return pkts;
+            }
+
+            SimTransformation state = transformState;
+
+            for (int i = 0; i < pkts.length; i++)
+            {
+                if (!RTCPPacketPredicate.INSTANCE.test(pkts[i]))
+                {
+                    continue;
+
+                }
+                if (RTCPHeaderUtils.getSenderSSRC(pkts[i]) != state.currentSSRC)
+                {
+                    // Anything from anywhere else gets axed.
+                    pkts[i] = null;
+                    continue;
+                }
+
+                RTCPIterator it = new RTCPIterator(pkts[i]);
+                while (it.hasNext())
```

Comment:
Hey, is this the loop which replaces fmj? Should we now nuke it from pom.xml? :)